### ISSUE TITLE
Heredoc not recognized after line continuation

### ISSUE THREAD

**Issue by jscheid** [hadolint/hadolint#1019]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Should accept heredocs when combined with line continuation (backslash at EOL.)

### Actual behavior

```
/tmp/Dockerfile:6:1 unexpected 'b' expecting '#', '\', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, a pragma, at least one space, or end of input
```

### Steps to reproduce the behavior

```sh
hadolint /tmp/Dockerfile
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.7-labs
FROM scratch

RUN foo && \
    bar <<BAR
blah
BAR
```

**Additional environment details (OS, stack version, etc.)**

**Comment by jscheid** [comment]
Oh there are plenty of workarounds of course, from putting it all onto one long line, to splitting up into multiple RUN layers (and then squashing the layers), to nested shell scripts like so (needs TAB for outer indentation I believe):

```Dockerfile
FROM ubuntu

RUN bash <<-SCRIPT
	echo foo && \
	cat <<-BAR
	  blah
	BAR
SCRIPT
```

**Comment by jscheid** [comment]
@electriquo this validates fine with `Haskell Dockerfile Linter 2.12.0`:

```Dockerfile
FROM ubuntu:20.04

RUN <<EOM cat
hadolint
EOM
```

**Comment by jscheid** [comment]
Thanks @m-ildefons!
